modules/storage-api/src/main/java/org/apache/ignite/internal/storage/engine/StorageEngine.java

```diff
@@ -58,4 +59,11 @@ default TableStorage createTable(TableConfiguration tableCfg) throws StorageExce
      * @throws StorageException If an error has occurs while creating the table.
      */
     MvTableStorage createMvTable(TableConfiguration tableCfg) throws StorageException;
+
+    /**
+     * Creates new transaction state storage for table.
+     *
+     * @return Transaction state storage.
+     */
+    TxnStateTableStorage createTxnStateTableStorage(TableConfiguration tableCfg);
```

Comment:
Tx state storage has nothing to do with the storage engine. Dude!

Reply:
Now wait a minute, how is it even possible? `TxnStateTableStorage` shouldn't be in the classpath of `storage-api` module. Have you changed dependencies? Does `storage-api` depend on `transactions` now? I don't understand